tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,544 @@
+
+                    var rgData = new ResourceGroupData(new AzureLocation(location));
+                    var rgLro = await subscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, _testResourceGroupName, rgData);
+                    _resourceGroup = rgLro.Value;
+                    _shouldDeleteResourceGroup = true;
+                }
+                else
+                {
+                    if (string.IsNullOrEmpty(resourceGroupName))
+                    {
+                        _skipTests = true;
+                        _skipReason = $"Missing required environment variable: {ResourceGroupEnvVar}";
+                        return;
+                    }
+
+                    _testResourceGroupName = resourceGroupName;
+                    _resourceGroup = await subscription.GetResourceGroups().GetAsync(resourceGroupName);
+                }
+
+                // Create unique store name for this test run
+                _testStoreName = $"integration-{Guid.NewGuid():N}".Substring(0, 20);
+
+                // Create the App Configuration store
+                var storeData = new AppConfigurationStoreData(new AzureLocation(location), new AppConfigurationSku("free"));
+                var createOperation = await _resourceGroup.GetAppConfigurationStores().CreateOrUpdateAsync(
+                    WaitUntil.Completed,
+                    _testStoreName,
+                    storeData);
+
+                _appConfigStore = createOperation.Value;
+                _appConfigEndpoint = new Uri(_appConfigStore.Data.Endpoint);
+
+                // Get the connection string for the store instead of using RBAC
+                var accessKeys = _appConfigStore.GetKeysAsync();
+                var primaryKey = await accessKeys.FirstOrDefaultAsync();
+
+                if (primaryKey == null)
+                {
+                    throw new InvalidOperationException("Failed to retrieve access keys from App Configuration store.");
+                }
+
+                _connectionString = primaryKey.ConnectionString;
+
+                // Initialize the configuration client with the connection string
+                _configClient = new ConfigurationClient(_connectionString);
+
+                // Add test settings to the store
+                foreach (var setting in _testSettings)
+                {
+                    await _configClient.SetConfigurationSettingAsync(setting);
+                }
+            }
+            catch (Exception ex)
+            {
+                _skipTests = true;
+                _skipReason = $"Failed to initialize integration tests: {ex.Message}";
+
+                // Clean up any partially created resources
+                await DisposeAsync();
+            }
+        }
+
+        /// <summary>
+        /// Cleans up the temporary App Configuration store after tests are complete.
+        /// </summary>
+        public async Task DisposeAsync()
```

Comment:
I think we should have Cleanup method that we call on intialization and tear down. That way we don't have stale stores.